### PR TITLE
[testnetv3-dev] Revision of Test_ShardSizeCalculator.cpp.

### PR DESCRIPTION
**Pull request by jendis**
## Description

Revision of Test_ShardSizeCalcucator.cpp based on #577.

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: tests/Utils/Test_ShardSizeCalculator.cpp [hunk 1]
```diff
@@ -17,131 +17,64 @@
...
-                      "Expected: 816. Result: " + to_string(result));
+typedef std::vector<uint32_t> TestDataSet;
+typedef std::map<uint32_t, TestDataSet> ShardSizeMap;
+
+void prepareTestdata(ShardSizeMap& testData) {
+  TestDataSet tds;
+  testData[651] = tds = {0, 1, 650, 651, 1000, 1367};
+  testData[684] = tds = {1368, 2131, 2132};
+  testData[711] = tds = {2133, 2200, 2867};
+  ;
```
Comment (ansnunez) [inline tests/Utils/Test_ShardSizeCalculator.cpp]: Extra semicolon

File: tests/Utils/Test_ShardSizeCalculator.cpp [hunk 2]
```diff
@@ -17,131 +17,64 @@
...
+  testData[717] = tds = {2868, 2869, 3674};
```
Comment (ansnunez) [inline tests/Utils/Test_ShardSizeCalculator.cpp]: Once libTestUtils is merged, maybe we can randomize the middle value within the range?

Comment (Jazzz42) [reply tests/Utils/Test_ShardSizeCalculator.cpp]: By my opinion randomizing the middle value is not necessary since this test focuses on the boundary values, so the middle value just confirms the "equivalence class" where the two boundaries belong. However randomizing the middle value would cause no harm so do as you like.

File: tests/Utils/Test_ShardSizeCalculator.cpp [hunk 3]
```diff
@@ -17,131 +17,64 @@
...
-      std::numeric_limits<uint32_t>::max() - 1);
-  BOOST_CHECK_MESSAGE(result == 819,
-                      "Expected: 819. Result: " + to_string(result));
-}
+#define TD_i td_i
+#define EXPECTED td_i.first
+#define NUMOFNODES_v td_i.second
 
-BOOST_AUTO_TEST_CASE(test_max_shards_exceed_upper_bound) {
+BOOST_AUTO_TEST_CASE(test_lower_bound) {
```
Comment (Jazzz42) [inline tests/Utils/Test_ShardSizeCalculator.cpp]: Testcase name "test_shard_size_bounds" seems more suitable to me.

Comment (jendis) [reply tests/Utils/Test_ShardSizeCalculator.cpp]: Agree, done.

File: tests/Utils/Test_ShardSizeCalculator.cpp [hunk 4]
```diff
@@ -17,131 +17,64 @@
...
+  testData[762] = tds = {6858, 7000, 7709};
+  testData[771] = tds = {7710, 8000, 8579};
+  testData[780] = tds = {8580, 9000, 9467};
+  testData[789] = tds = {9468, 10000, 10334};
+  testData[795] = tds = {10335, 11000, 14363};
+  testData[798] = tds = {14364, 15000, 15389};
+  testData[810] = tds = {15390, 16000, 18767};
+  testData[816] = tds = {18768, 19000, 20399};
+  testData[819] = tds = {20400, 300000, std::numeric_limits<uint32_t>::max()};
 }
```
[the remark this answers is not in the document]
Comment (jendis) [reply tests/Utils/Test_ShardSizeCalculator.cpp]: Agree, done.

File: tests/Utils/Test_ShardSizeCalculator.cpp (hunk 2 above)
Comment (jendis) [reply tests/Utils/Test_ShardSizeCalculator.cpp]: Agree, done.
